script/commands/register.ts

```diff
@@ -0,0 +1,149 @@
+import { Command } from 'commander';
+
+import { IgpHookType, config } from '../shared/config';
+import { typed, ContextHook } from '../shared/context';
+import { executeMultiMsg } from '../shared/contract';
+import { CONTAINER, Dependencies } from '../shared/ioc';
+
+export const registerCmd = new Command('register')
+  .description('Register new chain to contracts')
+  .configureHelp({ showGlobalOptions: true });
+
+registerCmd.command('igporacle').action(handleRegisterIgpOracle);
+registerCmd.command('ism-multisig').action(handleRegisterIsm);
+
+type AggregateHook = typed<'hpl_hook_aggregate'> & {
+  hooks: ContextHook[];
+};
+
+type IgpHook = typed<'hpl_igp'> & { oracle: typed<'hpl_igp_oracle'> };
+
+function isIgpHookType(hook: ContextHook): hook is IgpHook {
+  return hook.type === 'hpl_igp';
+}
+
+function findIsmsConfig() {
+  return config.deploy.ism;
+}
+
+function findIgpHookInAggregate() {
+  const defaultHook = config.deploy.hooks?.default;
+  if (defaultHook && defaultHook.type === 'aggregate') {
+    const igpHook = defaultHook.hooks.find(
+      (hook): hook is IgpHookType => hook.type === 'igp'
+    );
+    return igpHook;
+  }
+  return undefined;
+}
+
+async function handleRegisterIsm(_: object, cmd: Command) {
+  const { ctx, client } = CONTAINER.get(Dependencies);
+
+  const ismMultisig = ctx.deployments.isms;
+  let ismMultisigAddress: string | undefined;
+  if (ismMultisig && ismMultisig.type === 'hpl_ism_multisig') {
+    ismMultisigAddress = ismMultisig.address;
+  }
+  if (!ismMultisigAddress) {
+    throw new Error('Ism multisig context not found');
+  }
+  const ismsConfig = findIsmsConfig();
+  if (!ismsConfig) {
+    throw new Error('Ism multisig config not found');
+  }
+  switch (ismsConfig.type) {
+    case 'multisig': {
+      const multisig = {
+        type: 'hpl_ism_multisig',
+        address: ismMultisigAddress as string,
+      };
+      await executeMultiMsg(
+        client,
+        Object.entries(ismsConfig.validators).map(([domain, { addrs, threshold }]) => ({
```

Comment:
can we for sure override the same domain? I assume so, but just confirming

Reply:
double-checked and we can!